src/test/java/org/wildfly/extension/elytron/SslTestCase.java

```diff
@@ -0,0 +1,137 @@
+                        return null;
+                    }
+                    public void checkClientTrusted(X509Certificate[] certs, String authType) {}
+                    public void checkServerTrusted(X509Certificate[] certs, String authType) {}
+                }
+        };
+        SSLContext clientContext = SSLContext.getInstance("TLS");
+        clientContext.init(null, trustAllCerts, null);
+        return clientContext.getSocketFactory();
+    }
+
+    @Test
+    public void testSslService() throws Exception {
+        String subsystemXml = "<subsystem xmlns=\"" + ElytronExtension.NAMESPACE + "\">\n" +
+                "    <tls>\n" +
+                "        <keystores>\n" +
+                "            <keystore name=\"TestingCaKeyStore\" type=\"JKS\" password=\"123456\">\n" +
+                "                <file path=\"" + keyStoreFile + "\"/>\n" +
+                "            </keystore>\n" +
+                "            <keystore name=\"TestingCaTrustStore\" type=\"JCEKS\" password=\"1234567\">\n" +
+                "                <file path=\"" + trustStoreFile + "\"/>\n" +
+                "            </keystore>\n" +
+                "        </keystores>\n" +
+                "        <key-managers>\n" +
+                "            <key-manager name=\"MyKeyManager\" algorithm=\"SunX509\" keystore=\"TestingCaKeyStore\" password=\"12345678\"/>\n" +
+                "        </key-managers>\n" +
+                "        <trust-managers>\n" +
+                "            <trust-manager name=\"MyTrustManager\" algorithm=\"SunX509\" keystore=\"TestingCaTrustStore\"/>\n" +
+                "        </trust-managers>\n" +
+                "        <server-ssl-contexts>\n" +
+                "            <server-ssl-context name=\"ServerSslContext\" protocols=\"SSLv2 SSLv3 TLSv1 TLSv1_3 TLSv1_2 TLSv1_1\" key-managers=\"MyKeyManager\" trust-managers=\"MyTrustManager\"/>\n" +
+                //"            <server-ssl-context name=\"ClientSslContext\" protocols=\"SSLv2 SSLv3 TLSv1 TLSv1_3 TLSv1_2 TLSv1_1\" trust-managers=\"MyTrustManager\"/>\n" +
+                "        </server-ssl-contexts>\n" +
+                "    </tls>\n" +
+                "</subsystem>\n";
+
+        services = super.createKernelServicesBuilder(null).setSubsystemXml(subsystemXml).build();
+        if (!services.isSuccessfulBoot()) {
+            Assert.fail(services.getBootError().toString());
+        }
+
+        ServiceName serverServiceName = Capabilities.SSL_CONTEXT_RUNTIME_CAPABILITY.getCapabilityServiceName("ServerSslContext");
+        SSLContext serverSslContext = (SSLContext) services.getContainer().getService(serverServiceName).getValue();
+        Assert.assertNotNull(serverSslContext);
+        SSLServerSocketFactory serverSocketFactory = serverSslContext.getServerSocketFactory();
+
+        /*
+        ServiceName clientServiceName = Capabilities.SSL_CONTEXT_RUNTIME_CAPABILITY.getCapabilityServiceName("ClientSslContext");
```

Comment:
If this is not required in the first implementation of the test is there a reason to keep it?

If there is I would suggest adding a comment so in the future we know why it is commented out, if not probably better to just remove it now.